weave/ops_arrow/dict.py

```diff
@@ -406,3 +407,29 @@ def default_projection():
         ),
         table._artifact,
     )
+
+
+@op(
+    name="ArrowWeaveListTypedDict-keys",
+    input_type={"self": ArrowWeaveListType(types.TypedDict({}))},
+    output_type=lambda input_types: ArrowWeaveListType(types.List(types.String())),
+)
+def keys(self):
+    keys = list(self.object_type.property_types.keys())
+    return convert.to_arrow(
+        [keys] * len(self), types.List(types.List(types.String())), self._artifact
+    )
+
+
+@op(
+    name="ArrowWeaveListTypedDict-propertyTypes",
+    hidden=True,
+    input_type={"self": ArrowWeaveListType(types.TypedDict({}))},
+    output_type=lambda input_types: ArrowWeaveListType(types.String()),
+)
+def propertyTypes(self):
```

Comment:
This is a misnomer, you're not returning types. Can you call it columnNames?

Reply:
Fixed in next commit